lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_l10n_helper.rb

```diff
@@ -0,0 +1,178 @@
+        # @return [Hash<String, String>] A hash whose keys are the language codes (basename of `.lproj` folders) for which violations were found,
+        #         and the values are the output of the `diff` showing these violations.
+        #
+        def run(input_dir:, base_lang: DEFAULT_BASE_LANG)
+          check_swiftgen_installed || install_swiftgen!
+          find_diffs(input_dir: input_dir, base_lang: base_lang)
+        end
+
+        ##################
+
+        private
+        
+        # Path to the swiftgen binary installed at install_path
+        def swiftgen_bin
+          "#{install_path}/bin/swiftgen"
+        end
+
+        # Name to use for the generated files / output files of SwiftGen for each locale. Those files will be generated in the temporary directory to then diff them.
+        def output_filename(lang)
+          "L10nParamsList.#{lang}.txt"
+        end
+        
+        # The Stencil template that we want SwiftGen to use to generate the output.
+        # It iterates on every "table" (`.strings` file, in most cases there's only one, `Localizable.strings`),
+        # and for each, iterates on every entry found to print the key and the corresponding types parsed by SwiftGen from the placeholders found in that translation
+        def template_content
+          <<~TEMPLATE
+          {% macro recursiveBlock table item %}
+            {% for string in item.strings %}
+          "{{string.key}}" => [{{string.types|join:","}}]
+            {% endfor %}
+            {% for child in item.children %}
+            {% call recursiveBlock table child %}
+            {% endfor %}
+          {% endmacro %}
+
+          {% for table in tables %}
+          {% call recursiveBlock table.name table.levels %}
+          {% endfor %}
+          TEMPLATE
+        end
+
+        # Create the template file and the config file, in the temp dir, to be used by SwiftGen when parsing the input files.
+        #
+        # @return [(String, Array<String>)] A tuple of (config_file_absolute_path, Array<langs>)
+        #
+        def generate_swiftgen_config!(input_dir, output_dir)
+          # Create the template file
+          template_path = File.absolute_path(File.join(output_dir, 'strings-types.stencil'))
+          File.write(template_path, template_content)
+
+          # Dynamically create a SwiftGen config which will cover all supported languages
+          langs = Dir.chdir(input_dir) do
+              Dir.glob('*.lproj').map { |dir| File.basename(dir, '.lproj') }
+          end.sort
+      
+          config = {
+            'input_dir' => input_dir,
+            'output_dir' => output_dir,
+            'strings' => langs.map do |lang|
+              {
+                  'inputs' => ["#{lang}.lproj/Localizable.strings"],
+                  'options' => { 'separator' => "____" }, # Choose an unlikely one to avoid creating needlessly complex Stencil Context due to '.' in sentences sued as keys
```

Comment:
```suggestion
                  'options' => { 'separator' => "____" }, # Choose an unlikely one to avoid creating needlessly complex Stencil Context due to '.' in the sentences uses as keys
```

Reply:
Updated the whole comment in 5f9702350120f68002c0a79101e652a5a6610b11